### ISSUE TITLE
Review whether generated files need to be comitted

### ISSUE THREAD

**Issue by tombentley** [strimzi/strimzi-kafka-operator#1749]
As shown by #1747, it is confusing to new contributors that we commit (and check for) derived resources being committed to git. I don't recall the original motivation for this, but if files are generated by the build then there should be no need to commit them. 

Unless there's a solid reason otherwise, I think we should removed these derived files from git and remove the check.

**Comment by scholzj** [comment]
The generating of many of these files is too complicated and would often
mean that users need to build the project just to use it. Given it would
require Java, Maven and Make and not work properly on Windows, having these
files committed makes more sense to me from the overall simplicity
prespective.

**Comment by tombentley** [comment]
> would often mean that users need to build the project just to use it.

That's pretty normal for a repo on github. 

Who do we think is cloning the repo without the intention of building it?

**Comment by scholzj** [comment]
> Who do we think is cloning the repo without the intention of building it?

I assume all the users on Slack asking questions with latest are using it. You are saying that it is too complicated for developers to contribute because of this. But:
a) This will force all users to actually do the same just to use master
b) Devs will still need to do it to test their changes properly.

**Comment by tombentley** [comment]
OK then.

I wonder if `git add`ing the generated files as part of the `make` build would help, or whether new devs would it that confusing that stuff was in their index which they'd not added themselves.

**Comment by scholzj** [comment]
> I wonder if git adding the generated files as part of the make build would help, or whether new devs would it that confusing that stuff was in their index which they'd not added themselves.

You mean that the `make` or `mvn` commands would automatically do `git add`?

**Comment by tombentley** [comment]
Yes. `make` would be easier, but I guess it wouldn't catch people doing a pure maven build. I guess we could use the mvn exec plugin to do it via maven though. **If** it's not too weird.

**Comment by scholzj** [comment]
I think we will need to be careful so that it doesn't add some unwanted files etc. But that hould be ok I guess. Also, it remains to be seen whether people don't run the calls today (e.g. `make helm_install` for the PR which triggered this issue) or don't commit the files. My vote is for the first one in which case doing `git add` there would not help.
